internal/apis/kfd/v1alpha2/ekscluster/common/distribution.go

```diff
@@ -353,5 +355,13 @@ func (d *Distribution) extractARNsFromTfOut() (map[string]string, error) {
 		}
 	}
 
+	awsRoute53ZonePublicID, ok := out["aws_route53_zone_public_id"]
+	if ok {
+		arns["aws_route53_zone_public_id"], ok = awsRoute53ZonePublicID.Value.(string)
+		if !ok {
+			return nil, errTerraformOutputToStr
```

Comment:
if we are returning this error only in this case, you should be more specific in the error text, saying that the Zone could not be found or something like that intead of error casting terraform output to string.